#163542246: Register a user as a candidate

Hoxtygen opened the pull request:
#### What does this PR do?
This endpoint ensures that the admin can register a user who shows
interest in running for an office as a candidate for that office. It
also ensures that a candidate does not apply for the same office twice

#### Description of Task to be completed?
Have the following endpoint working
`POST /api/v1/office/<office-id>/register`

#### How should this be manually tested?
clone the repo and cd into it then on the terminal do `npm run dev`
Using postman, test the endpoint with the following header information

    key: Content-Type value: application/json


#### What are the relevant pivotal tracker stories?
[Delivers [#163542246](https://www.pivotaltracker.com/story/show/163542246)]

#### Screenshots (if appropriate)
![registercandidate](https://user-images.githubusercontent.com/19408669/52114705-07df0f00-260d-11e9-9dfa-40a3616c7023.PNG)

Hoxtygen committed:
ft(feature):Register a user as a candidate

This endpoint ensures that the admin can register a user who shows
interest in running for an office as a candidate for that office. It
also ensures that a candidate does not apply for the same office twice

[Delivers #163542246]

Orlayhemmy commented on server/controllers/candidateOfficeController.js:
```diff
@@ -0,0 +1,35 @@
+    console.log(req.body);
+    console.log(req.params.id);
+    dbConfig.query('INSERT INTO politico_andela.candidates (office, candidate) VALUES ($1, $2) RETURNING *', [office, candidate])
+      .then((candi) => {
```
Use a better variable name as compared to `candi`

Hoxtygen replied:
Oh! that was used in hindsight... Changing it..

Hoxtygen replied:
fixed.

Hoxtygen committed:
refactor: change variable name

This commit renames a variable to give it a more semantic meaning.

[Fixes #163542246]
